### PR TITLE
Add config "DataDisk" to specify the default Longhorn partition

### PR DESCRIPTION
**Pull request by johnliu55tw**
- Refactor: Move partition layout logic into cos.go
- Upgrade `yip` to 0.9.25 so we could use the latest `yip/schema` package to define the "layout" stage
- Add config "DataDisk" to specify the default Longhorn disk

## Test plan
1. Prepare a machine with two disks, let's say they will be `/dev/vda` and `/dev/sda` during the installation.
2. During the installation, select `/dev/vda` as the OS disk
3. At the step "Remote config", provide the following config to use `/dev/sda` as the default Longhorn disk:
    ```
    install:
      data_disk: /dev/sda
    ```
4. After the installation finished, wait for the system to reboot and verify the following:
    - `/dev/sda` is mounted to `/var/lib/longhorn_data` as the default disk. Create a VM and verify Longhorn is storing data into `/var/lib/longhorn_data`. You can create a huge file in the VM and see the space of `/dev/vda` is getting consumed.
    - `/var/lib/longhorn` still exists, and Longhorn binaries are stored here.
        ```
        $ ls -l /var/lib/longhorn/engine-binaries
        total 4
        drwxr-xr-x 2 root root 4096 Jan 19 08:14 longhornio-longhorn-engine-v1.2.3
        ```
5. Prepare another machine to join the cluster. Again you need two disks on this one.
6. Provide the same config as step 3 to use the other disk as the default Longhorn disk.
7. Perform the same verification as step 4 on this join Node.

## Note
- Will create another PR to add `VmDataDisk` config to installer UI.
- For the upgrade to adopt this change, we also need https://github.com/harvester/harvester/pull/1832

## Relates issues
- https://github.com/harvester/harvester/issues/1728

### CODE REVIEW

File: package/harvester-os/files/usr/sbin/harv-install [hunk 1]
```diff
@@ -2,6 +2,22 @@
...
+    # Also, while yip is partitioning the disk, if it sees the LABEL to be used exists,
+    # it won't create the partition. So it's necessary to clear the label
+    for part in $(blkid -t LABEL="$DATA_DISK_FSLABEL" -o device); do
+        echo "Remove filesystem label from $part"
+        # Run this tune2fs twice because sometimes the first run would show "Recovering journal"
+        # and label is not modified
+        tune2fs -L "" $part > /dev/null
+        tune2fs -L "" $part > /dev/null
+    done
+    udevadm settle
```
[the remark this answers is not in the document]
Comment (johnliu55tw) [reply package/harvester-os/files/usr/sbin/harv-install:20]: Yeah, it's possible for such asynchronous operations to block indefinitely. So by default, `udevadm settle` has a timeout of 120 seconds.

File: pkg/console/install_panels.go [hunk 2]
```diff
@@ -1419,6 +1419,15 @@ func addInstallPanel(c *Console) error {
 				c.config.NoDataPartition = true
 			}
 
+			// If DataDisk is specified, NoDataPartition should be set to "false" as we will have
+			// VM data partition, just on other disk.
+			if c.config.DataDisk != "" {
+				msg := fmt.Sprintf("Use %s as default VM data disk", c.config.DataDisk)
+				printToPanel(c.Gui, msg, installPanel)
+				logrus.Info(msg)
+				c.config.NoDataPartition = false
```
[the remark this answers is not in the document]
Comment (johnliu55tw) [reply pkg/console/install_panels.go:1428]: I'm more towards the idea of "`NoDataPartition` config is ignored if you chose another disk as the default data disk", and we add a note to the doc explaining this idea.

**Review by weihanglo — APPROVED** [review]
LGTM

- [x] Data disk contains `HARV_LH_LABEL`
- [x] Backing images and longhorn replicas are on data disk
- [x] Ther are only engine-binaries under `/var/lib/longhorn`

**Review by bk201 — APPROVED** [review]
LGTM

**Pull request merged**